packages/react-composites/src/composites/CallComposite/utils/MediaGalleryUtils.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { _formatString } from '@internal/acs-ui-common';
+import { RemoteParticipantState } from '@internal/calling-stateful-client';
+import { useMemo, useState } from 'react';
+import { useLocale } from '../../localization';
+import { useSelector } from '../hooks/useSelector';
+import { getRemoteParticipantsConnectedSelector } from '../selectors/mediaGallerySelector';
+
+type ParticipantChangedAnnouncmentStrings = {
+  participantJoinedNoticeString: string;
+  twoParticipantJoinedNoticeString: string;
+  threeParticipantJoinedNoticeString: string;
+  participantLeftNoticeString: string;
+  twoParticipantLeftNoticeString: string;
+  threeParticipantLeftNoticeString: string;
+  unnamedParticipantString: string;
+  manyParticipantsJoined: string;
+  manyParticipantsLeft: string;
+  manyUnnamedParticipantsJoined: string;
+  manyUnnamedParticipantsLeft: string;
+};
+
+/**
+ * sets the announcement string whenever a Participant comes or goes from a call to be
+ * used by the system narrator.
+ *
+ * @returns string to be used by the narrator and Announcer component
+ *
+ * @internal
+ */
+export const useParticipantChangedAnnouncement = (): string => {
+  const locale = useLocale().strings.call;
+  const strings = useMemo(() => {
+    return {
+      participantJoinedNoticeString: locale.participantJoinedNoticeString,
+      twoParticipantJoinedNoticeString: locale.twoParticipantJoinedNoticeString,
+      threeParticipantJoinedNoticeString: locale.threeParticipantJoinedNoticeString,
+      participantLeftNoticeString: locale.participantLeftNoticeString,
+      twoParticipantLeftNoticeString: locale.twoParticipantLeftNoticeString,
+      threeParticipantLeftNoticeString: locale.threeParticipantLeftNoticeString,
+      unnamedParticipantString: locale.unnamedParticipantString,
+      manyParticipantsJoined: locale.manyParticipantsJoined,
+      manyParticipantsLeft: locale.manyParticipantsLeft,
+      manyUnnamedParticipantsJoined: locale.manyUnnamedParticipantsJoined,
+      manyUnnamedParticipantsLeft: locale.manyUnnamedParticipantsLeft
+    };
+  }, [locale]);
+  const [announcerString, setAnnouncerString] = useState<string>('');
+  const currentParticipants = useSelector(getRemoteParticipantsConnectedSelector);
+  const [previousParticipants, setPreviousParticipants] = useState<RemoteParticipantState[]>(currentParticipants);
+
+  const resetAnnoucement = (string: string): void => {
+    // Is this really needed?
+    // React does not promise that we will see a DOM update with '' value.
+    // If we're getting announcer to announce the same name twice in sequence, it's probably keyed off of the
+    // actual DOM node, which will update even if we don't reest to '' first.
+    setAnnouncerString('');
```

Comment:
resolve comment about this being removed, given two setStates execute consecutively there is an incredibly small chance there would ever be a render pass between them